src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -32,15 +32,19 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, BindingD
             }
 
             MethodSymbol? cloneMethod = null;
-            if (!receiverType.IsErrorType())
+            if (receiverType.IsValueType)
```

Comment:
It feels like for enums for example there is nothing you can assign in the right side of the `with`. Is there any item tracking giving a warning when the right side of the `with` is empty?

Reply:
I have a couple of tests with an empty initializers in `with`. I don't think that should be a warning, as empty initializers are also allowed in object creation.

---
In reply to: [608246244](https://github.com/dotnet/roslyn/pull/52319#discussion_r608246244) [](ancestors = 608246244)